### ISSUE TITLE
first-timers-only: Add test to make sure resultNoMatch callbacks returns possible sentences

### ISSUE THREAD

**Issue by TalAter** [TalAter/annyang#168]
#### first-timers-only

This issue is tagged :octocat: **first-timers-only**. It is only for people who have never contributed to open source before, and are looking for an easy way take their first steps.

Thank you for your help :heart:

#### Background

Callbacks added using `annyang.addCallback('resultMatch', aFunction)` will be called when a sentence a user said does match any of the currently registered commands (see [docs](https://github.com/TalAter/annyang/tree/master/docs#addcallbacktype-callback-context)).

The callback function should be called with an array of the possible sentences the user may have said as the first argument.

#### The Issue

The test specs do not currently test to verify that the callback function is called with the possible sentences the user may have said

Add a test spec that verify this spec:
````javascript
  it('should call the callback with the the first argument containing an array of all possible Speech Recognition Alternatives the user may have said', func1);
}
````
This should be added into the following test suite:
````javascript
describe("annyang.addCallback('resultNoMatch')", function() {
````

The last test described inside `describe("annyang.addCallback('resultMatch')", function() {…});` in *test/spec/BasicSpec.js*, can be used as a template for adding the new spec described above.

#### Contributing to annyang

- [ ] Fork the repository from the [annyang GitHub page](https://github.com/TalAter/annyang).
- [ ] Clone a copy to your local machine with `$ git clone git@github.com:YOUR-GITHUB-USER-NAME/annyang.git`
- [ ] Make sure you have *node.js* and *npm* installed on your machine. You can use this [guide](https://docs.npmjs.com/getting-started/installing-node) for help.
- [ ] Install all of annyang's development dependencies with npm. `$ cd annyang; npm install`
- [ ] Run grunt to make sure everything runs smoothly `$ grunt`
- [ ] Code, code, code.
- [ ] Run `$ grunt` after making changes to verify that everything still works and run the build process. Alternatively, you can just run `$ grunt watch` once, and leave that process running. It will continuously run all the tests and build the files every time you make a change to one of annyang's files. It will even *beep* if you make an error, and help you debug it.:+1:
- [ ] Before committing your changes, the last step must always be running `$ grunt`. This makes sure everything works, and all files are kept up to date with your changes.
- [ ] Once you've made sure all your changes work correctly, and committed all your changes, push your local changes back to github with `$ git push -u origin master`
- [ ] Visit your fork on GitHub.com ([https://github.com/YOUR-USER-NAME/annyang](https://github.com/YOUR-USER-NAME/annyang)) and create a pull request for your changes.
- [ ] Makes sure your pull request describes exactly what you changed and references this issue (just include the issue number in the title like this: #49)

#### About the tests

annyang is tested using [Jasmine](http://jasmine.github.io/2.0/introduction.html).

BasicSpec.js contains a series of spec groups (e.g. `describe('a spec group', function() {})`) which each contain 1 or more specs (e.g. `it('should do stuff', function() {})`). Some of the spec groups also contain some code which runs before each spec (`beforeEach(function() {})`).

To simulate Speech Recognition in the testing environment, annyang uses a mock object called [Corti](https://github.com/TalAter/Corti) which mocks the browser's SpeechRecognition object. Corti also adds a number of utility functions to the SpeechRecognition object which simulate user actions (e.g. `say('Hello there')`), and allow checking the SpeechRecognition's status (e.g. `isListening() === true`).

**Comment by timotius02** [comment]
Trying to tackle this issue now. I'm just wondering why annyang is adding the words "and so on" for each possible sentence.

**Comment by TalAter** [comment]
Hi @timotius02 , thanks for pitching in! :fist: 

That is actually added by [Corti](https://github.com/TalAter/Corti), the mock object annyang's tests use to simulate speech recognition in a testing environment.

Because you don't have support for Speech Recognition in the command line, or in the headless browser used by the tests, I created a mock object called [Corti](https://github.com/TalAter/Corti) which mocks the browser's SpeechRecognition functionality.

Corti duplicates the behaviour of many of SpeechRecognition's features, as well as adding a number of utility functions to the SpeechRecognition object which simulate user actions (e.g. `recognition.say('Hello there')`), instead of speaking into the microphone and `recognition.isListening()` to check SpeechRecognition's status (instead of the blinking red dot in the browser.

Since `recognition.say()` doesn't actually use real Speech Recognition, I simulated different speech recognition alternatives by taking the original sentence and adding "and so on", "and so on" to each alternative.
